.github/workflows/e2e-test.yml

```
@@ -59,47 +63,57 @@ jobs:
 
       - name: Test aether-pingall
         working-directory: aether-onramp
-        run: make aether-pingall
+        run: |
+          set -e
+          make aether-pingall
 
       - name: Install aether-k8s
         working-directory: aether-onramp
         run: |
+          set -e
           make aether-k8s-install
 
       - name: Install 5G Core
         working-directory: aether-onramp
         run: |
+          set -e
           make aether-5gc-install
 
       - name: Patch K8s Deployment to attach local image
         run: |
-          set -x
-          git clone https://github.com/omec-project/amf && \
-          cd amf && \
-          PR_NUMBER=$(echo "${GITHUB_REF}" | awk -F'/' '{print $3}') && \
-          git fetch origin pull/${PR_NUMBER}/merge:pr-${PR_NUMBER} && \
+          set -ex
+          REPO_NAME=$(echo "${GITHUB_REPOSITORY}" | cut -d'/' -f2)
+          REPO_URL="https://github.com/omec-project/${REPO_NAME}"
+          git clone ${REPO_URL} ${REPO_NAME} && cd ${REPO_NAME}
+          PR_NUMBER=$(echo "${GITHUB_REF}" | awk -F'/' '{print $3}')
+          git fetch origin pull/${PR_NUMBER}/merge:pr-${PR_NUMBER}
           git checkout pr-${PR_NUMBER}
-          docker build -t amf:testing . || exit 1
-          kubectl set image deployment/amf amf=amf:testing -n aether-5gc || exit 1
-          kubectl -n aether-5gc patch deployment amf -p '{"spec":{"template":{"spec":{"containers":[{"name":"amf","imagePullPolicy":"Never"}]}}}}' || exit 1
+          if [ "${REPO_NAME}" = "webconsole" ]; then
+            REPO_NAME="webui"
+          fi
+          docker build -t ${REPO_NAME}:testing .
+          kubectl set image deployment/${REPO_NAME} ${REPO_NAME}=${REPO_NAME}:testing -n aether-5gc
+          kubectl -n aether-5gc patch deployment ${REPO_NAME} -p '{"spec":{"template":{"spec":{"containers":[{"name":"${REPO_NAME}","imagePullPolicy":"Never"}]}}}}'
```

```suggestion
    kubectl -n aether-5gc patch deployment ${REPO_NAME} \
    -p "{\"spec\":{\"template\":{\"spec\":{\"containers\":[{\"name\":\"${REPO_NAME}\",\"image\":\"${REPO_NAME}:testing\",\"imagePullPolicy\":\"Never\"}]}}}}"
```
Single quotes does not allow to sue the REPO_NAME variable. We need to use double quotes here.

---

The suggested change is tested here: https://github.com/omec-project/amf/pull/427

---

Thanks for your input. I am 99% certain I tested a solution similar to your but it did not work. Anyway, I am going to give it a try.
BTW, your [PR AMF#427](https://github.com/omec-project/amf/pull/427) is still using the "old" change (single quotation marks) as you can see it here: https://github.com/omec-project/amf/actions/runs/14265353107/job/39985831972?pr=427#step:10:399

---

Thanks for your input. It worked

---

I do not know why workflow logs shows  single quotes but in fact we are using double quotes to replace the variable. Otherwise, it will not work. Single quotes does not allow parameter replacement.